File: Arrays/Python/SmallestLargest.py
# Find smallest & largest element from the array along with their indexes.
def find_smallest(array):
    smallest = array[0]
    for i in range(1, len(array)):
        if array[i] < smallest:
            smallest = array[i]
    return smallest

def find_largest(array):
    largest = array[0]
    for i in range(1, len(array)):
        if array[i] > largest:
            largest = array[i]
    return largest

def find_index(array, element):
    for i in range(len(array)):
        if array[i] == element:
            return i
    return -1

def main():
    array = [-15, 56, 90, 23, -23]
    smallest = find_smallest(array)
    largest = find_largest(array)

    print(f"Smallest element is {smallest} at index {find_index(array, smallest)}")
    print(f"Largest element is {largest} at index {find_index(array, largest)}")

File: Arrays/C/SmallestLargest.c
// Find smallest & largest element from the array along with their indexes.
#include <stdio.h>

int findSmallest(int array[], int size) {
    int smallest = array[0];
    for (int i = 1; i < size; i++) {
        if (array[i] < smallest)
            smallest = array[i];
    }
    return smallest;
}

int findLargest(int array[], int size) {
    int largest = array[0];
    for (int i = 1; i < size; i++) {
        if (array[i] > largest)
            largest = array[i];
    }
    return largest;
}

int findIndex(int array[], int size, int element) {
    for (int i = 0; i < size; i++) {
        if (array[i] == element)
            return i;
    }
    return -1;
}

void SmallestLargest() {
    int array[5] = {-15, 56, 90, 23, -23};
    int smallest = findSmallest(array, 5);
    int largest = findLargest(array, 5);

    printf("Smallest element is %d at index %d\n", smallest, findIndex(array, 5, smallest));
    printf("Largest element is %d at index %d\n", largest, findIndex(array, 5, largest));
}

int main() {
    SmallestLargest();
    return 0;
}